src/box/replication.cc

```diff
@@ -206,6 +210,71 @@ replicaset_set_sync_quorum(void)
 	replication_sync_quorum_auto = replicaset.applier.connected;
 }
 
+/**
+ * Deactivates all expired anonymous replicas. They are expected to be
+ * deleted on deactivation as orphan.
+ * Returns lowest time-to-live among remaining anonymous replicas via
+ * `lowest_ttl`. If there are no anonymous replicas left or the function has
+ * failed to deactivate a replica, `replication_anon_gc_timeout` is returned.
+ */
+static void
+replication_anon_gc(double *lowest_ttl)
```

Comment:
Why is this done in `replication.cc` and not in `gc_cleanup_fiber`. It seems the latter is the better place for this, we anyway don't need to remove consumers precisely after timeout and gc fiber's waken up every `gc_advance`, very often

Reply:
Conceptually, it's garbage collector for anonymous replicas themselves, not only their WAL GC consumers.

Reply:
Hm. I think some confusion is coming from the option name - `replication_anon_gc_timeout`. It makes us think like it is about GC state of the anon replicas, not about the anon replicas themselves. Perhaps we could remove the uncertainty by naming it `replication_anon_deactivation_timeout`?

Reply:
>  I think some confusion is coming from the option name - `replication_anon_gc_timeout`.

Yeah, since we call WAL GC as GC, any abbreviation of "garbage collector" makes us think about xlogs if it's somehow related to it :persevere:

I've asked Vlad Grubov and I'm waiting for his answer. Now I suggest an alternative name - `replication_anon_ttl`.

Reply:
Renamed to `replication_anon_ttl`.